code/datums/crew_objective.dm

```diff
@@ -307,11 +330,161 @@ ABSTRACT_TYPE(/datum/objective/crew/janitor)
 					return 0
 			return 1
 
-//	bartender
+#define DRINK_OBJ_COUNT 3
+ABSTRACT_TYPE(/datum/objective/crew/bartender)
+/datum/objective/crew/bartender
+	shotgun
+		explanation_text = "Don't lose your shotgun!"
+		check_completion()
+			if(owner.current && owner.current.check_contents_for(/obj/item/gun/kinetic/riotgun))
+				return TRUE
+			else
+				return FALSE
+	drinks
+		var/completed = FALSE
+		var/ids[DRINK_OBJ_COUNT]
+		New()
+			..()
+			var/list/cocktails = concrete_typesof(/datum/reagent/fooddrink/alcoholic)
+			var/list/blacklist = list(
```

Comment:
this can be static and on the type level

Reply:
Fixed